core/templates/pages/exploration-editor-page/services/user-exploration-permissions.service.ts

```diff
@@ -42,6 +42,13 @@ export class UserExplorationPermissionsService {
     }
     return UserExplorationPermissionsService.permissionsPromise;
   }
+
+  async fetchPermissionsAsync(): Promise<ExplorationPermissions> {
+    UserExplorationPermissionsService.permissionsPromise = (
+      this.explorationPermissionsBackendApiService.getPermissionsAsync());
```

Comment:
Will this fetch the permissions again if we already have them in the cache?

Reply:
Yes

Reply:
I'm not sure, but can you please re-verify? [Maybe I'm wrong, but wanted to get a confirmation before taking another pass!]

Reply:
Ah, I guess I wasn't clear enough.
1. https://github.com/oppia/oppia/blob/96a08a269aff953208e31b90ffcba074629298ad/core/templates/pages/exploration-editor-page/services/user-exploration-permissions.service.ts#L47-L48 fetches from the backend.
2.  https://github.com/oppia/oppia/blob/96a08a269aff953208e31b90ffcba074629298ad/core/templates/pages/exploration-editor-page/services/user-exploration-permissions.service.ts#L38-L44 fetches from the backend if permissions is not present in the cache

3. https://github.com/oppia/oppia/blob/96a08a269aff953208e31b90ffcba074629298ad/core/templates/pages/exploration-editor-page/services/user-exploration-permissions.service.ts#L46-L52 fetches from backend irrespective of whether it is present in the cache or noe

Reply:
I see, thanks for the explanation.